tests/base/src/test/java/org/wildfly/security/audit/SyslogAuditEndpointTest.java

```diff
@@ -0,0 +1,331 @@
+     */
+    @BeforeClass
+    public static void initializeServer() throws Exception {
+        socketFactory = SocketFactory.getDefault();
+        udpServer = SimpleSyslogServer.createUdp(UDP_PORT);
+        tcpServer = SimpleSyslogServer.createTcp(TCP_PORT, false);
+    }
+
+    /**
+     * Safely shutdowns the server
+     */
+    @AfterClass
+    public static void shutdownServer() {
+        udpServer.close();
+        tcpServer.close();
+    }
+
+    /**
+     * Tests the syslog audit endpoint with RFC 3164 format and critical priority
+     */
+    @Test
+    public void testGenericRfc3164SyslogEvent() throws Exception {
+        AuditEndpoint endpoint = setupEndpointUdp(SyslogHandler.SyslogType.RFC3164);
+        endpoint.accept(EventPriority.CRITICAL, RFC3164_LOG_MESSAGE);
+        endpoint.close();
+        discardMessage(udpServer);
+        verifyUdpMessage(RFC3164_LOG_MESSAGE);
+    }
+
+    /**
+     * Tests the syslog audit endpoint with RFC 5424 format and critical priority
+     */
+    @Test
+    public void testGenericRfc5424SyslogEvent() throws Exception {
+        AuditEndpoint endpoint = setupEndpointUdp(SyslogHandler.SyslogType.RFC5424);
+        endpoint.accept(EventPriority.CRITICAL, RFC5424_LOG_MESSAGE);
+        endpoint.close();
+        discardMessage(udpServer);
+        verifyUdpMessage(RFC5424_LOG_MESSAGE);
+    }
+
+    /**
+     * Tests that the endpoint rejects a bad integer
+     */
+    @Test
+    public void testBadIntegerReconnectAttempts() throws Exception {
+        try {
+            // Since the endpoint gives a message upon creation there is no need to call .accept()
+            AuditEndpoint endpoint = setupEndpointBadUdpReconnectAttempts(BAD_RECONNECT_NUMBER);
+            Assert.fail("No error was thrown while attempting to log with an invalid reconnect-attempts number.");
+        } catch (IllegalArgumentException e) {
+            assertCorrectError(e, new String[] {"ELY11008", Integer.toString(BAD_RECONNECT_NUMBER)});
+        }
+    }
+
+    /**
+     * Tests that the server continuously attempts to resend the message for {@link SyslogAuditEndpointTest#RECONNECT_TIMEOUT} milliseconds
+     */
+    @Test
+    public void testFailureInfiniteReconnectAttempts() throws Exception {
+        // Windows Server can have an issue echoing back an error, causing the test to fail by not seeing a failure sending
+        Assume.assumeFalse("Test does not run on Windows Server", SystemUtils.OS_NAME.contains("Windows Server"));
+        Thread t = new Thread(() -> {
+            try {
+                // Since the endpoint gives a message upon creation there is no need to call .accept()
+                AuditEndpoint endpoint = setupEndpointBadUdpReconnectAttempts(-1);
+                if (!Thread.currentThread().isInterrupted()) {
+                    this.failString = "No error was thrown while attempting to log to a bad IP with UDP.";
```

Comment:
I don't see `failString` used in any other methods. Is there a reason why it isn't a method local variable?

Reply:
failString can't be a method local variable for both the child thread lambda and the parent thread can access it (from my understanding, this has to due with JVM copying the local variable vs passing a reference to it for the instance variable when creating the lambda)

Reply:
Thank you for the explanation!